rooms.js

```diff
@@ -163,6 +165,107 @@ var Room = (function () {
 		}
 		return true;
 	};
+	//mute handling
+	Room.prototype.runMuteTimer = function () {
+		if (this.muteTimer || this.mutedUsers.length === 0) return;
+		var timeUntilExpire = this.mutedUsers[0].time - Date.now();
+		if (timeUntilExpire <= 0) {
+			var user = Users(this.mutedUsers[0].userid);
+			if (user) {
+				user.popup("Your mute has expired.");
+				this.unmute(user);
+			} else {
+				this.mutedUsers.splice(0, 1);
+				this.runMuteTimer();
+			}
+		} else {
```

Comment:
With the while-loop approach I'm talking about, you don't need to repeat the `.popup("Your mute has expired.");` etc code since it always happens at the top of `runMuteTimer`.

That's what I meant when I said your code was brittle. Your unmute code is in two places when it only needs to be in one place in a `while`-loop at the top of `runMuteTimer`.